src/game_api/search.cpp

```diff
@@ -2034,6 +2034,50 @@ std::unordered_map<std::string_view, AddressRule> g_address_rules{
             .offset(0x21)
             .at_exe(),
     },
+    {
+        "unload_layer"sv,
+        // bp on destroy entity, leave level, it's third in stack or something
+        PatternCommandBuffer{}
+            .find_inst("49 89 cc 8b 41 18 85 c0 74 0b 49 8b 74 24 08"_gh)
+            .at_exe()
+            .function_start(),
+    },
+    {
+        "init_layer"sv,
+        // TODO
+        PatternCommandBuffer{}
+            .from_exe_base(0x228b58f0),
+    },
+    {
+        "dead_players"sv,
+        // TODO
+        PatternCommandBuffer{}
+            .from_exe_base(0x22c061d0),
+    },
+    {
+        "spawn_transition"sv,
+        // TODO
+        PatternCommandBuffer{}
+            .from_exe_base(0x22afe5c0),
```

Comment:
That's theme virtual 21

Reply:
Same for the rest of transitions, just different theme virtual, there are 5:
standard, jungle and volcana (have some extra code before jumping to the standard function),
Co duat and hundun use the same one, there is also basecamp one, not sure for what for, and city of gold (probably deciding which one to call, standard or to duat)

Reply:
I know they're all theme virtual 21, and I specifically didn't hook the vtable for all themes (like in LevelGenSystem::hook_theme) cause that didn't work. I don't know why, but get_entities() was empty in the callbacks made this way, even though it worked fine when using state.current_theme:set_post_spawn_transition()

Reply:
And obviously this callback would be useless if you can't find the entities just spawned...

Reply:
> I know they're all theme virtual 21, and I specifically didn't hook the vtable for all themes (like in LevelGenSystem::hook_theme) cause that didn't work. I don't know why, but get_entities() was empty in the callbacks made this way, even though it worked fine when using state.current_theme:set_post_spawn_transition()

That's super weird, you sure you didn't made off by one error or something? or maybe state local is busted when initiating level?
If not, maybe we can use `get_virtual_function_address` or something like that here, instead of harcoded offset

Reply:
It was pretty late then, but after testing again I am still pretty confident I'm hooking the right function, cause it still doesn't work if I call current_theme:spawn_transition() directly. Sounds like get_virtual_function_address would simplify this though.